app/services/umrdr/date_coverage_service.rb

```diff
@@ -0,0 +1,130 @@
+require 'date'
+require 'edtf'
+
+module Umrdr
+  class DateCoverageService  
+
+    # EDTF's notation for interval begin and end when unspecified.
+    EDTF_OPEN_BEGIN = :unknown
+    EDTF_OPEN_END = :open
+
+    class << self
+      # Given a parameters hash with date coverage keys
+      # Returns an EDTF Interval or nil.
+      def params_to_interval( params )
+        begin_date = make_begin_date(
+                      integerize(params[:date_coverage_begin_year]),
+                      integerize(params[:date_coverage_begin_month]),
+                      integerize(params[:date_coverage_begin_day])
+        )
+        end_date = make_end_date(
+                      integerize(params[:date_coverage_end_year]),
+                      integerize(params[:date_coverage_end_month]),
+                      integerize(params[:date_coverage_end_day])
+        )
+        EDTF::Interval.new(begin_date, end_date) unless begin_date == :unknown && end_date == :open
+      end
+
+      # Given an EDTF Interval
+      # Returns a corresponding hash of parameters for an edit form.
+      def interval_to_params( interv )
+        return unless interv.is_a? EDTF::Interval
+        reverse_interval!(interv) if interval_reversed?(interv) 
```

Comment:
Currently this is a strategy of switching the begin and end of an interval if the beginning is later than the given end? This is also an upstream issue in ruby-edtf. 

I think the options are:
1. "correct" the reversed interval.  I see that we'd have to do this in params_to_interval as well.
2. abort or error if given a reversed interval.  This would also have to be done in params_to_interval.

The 2nd option is what I'm leaning towards the more I think about it.  What do you think about each?